src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacExtensionDefaultAttribute/DacExtensionDefaultAttributeAnalyzer.cs

```diff
@@ -1,153 +1,269 @@
+
+            var (pxDefaultAttribute, hasPersistingCheckNothing) =
+                GetPXDefaultInfo(pxContext, attributeInformation, attributes, cancellationToken);
+            if (pxDefaultAttribute == null || hasPersistingCheckNothing)
+            {
+                return null;
+            }
+
+            
+            var baseExtensionAndDacTypes = dacExtension.GetDacExtensionsWithDac(pxContext).Skip(1);
+
+            // If Dac extension contains PXDefaultAttribute without PersistingCheck.Nothing
+            // we need to look to attribute wich it overrides:
+            // if base attribute is also doesn't contain PersistingCheck.Nothing it is legitimately
+            foreach (var darOrExtension in baseExtensionAndDacTypes)
+            {
+                var declaredProperty = GetDeclaredPropertyFromTypeByName(darOrExtension, propertyName);
+                if (declaredProperty == null)
+                {
+                    continue;
+                }
+
+                var attributesBase = declaredProperty.GetAttributes();
+                var (pxDefaultAttributeBase, hasPersistingCheckNothingBase) =
+                    GetPXDefaultInfo(pxContext, attributeInformation, attributesBase, cancellationToken);
+
+                if (pxDefaultAttributeBase == null)
+                {
+                    continue;
+                }
+
+                return hasPersistingCheckNothingBase ?
+                    pxDefaultAttribute :
+                    null;
+            }
+
+            return pxDefaultAttribute;
+        }
+
+        private static (AttributeData PXDefaultAttribute, bool HasPersistingCheckNothing) GetPXDefaultInfo (
+            PXContext pxContext, AttributeInformation attributeInformation,
+            ImmutableArray<AttributeData> attributes, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var pxDefaultAttribute = GetPXDefaultAttribute(pxContext, attributeInformation, attributes, cancellationToken);
+            if (pxDefaultAttribute == null)
+            {
+                return (null, false);
+            }
+
+            var hasPersistingCheckNothing = pxDefaultAttribute.NamedArguments
+                .Where(na => nameof(PXDefaultAttribute.PersistingCheck).Equals(na.Key, StringComparison.Ordinal))
+                .Select(na => na.Value.Value)
+                .Where(v => v is int persistingCheck && persistingCheck == (int)PXPersistingCheck.Nothing)
+                .Any();
+
+            return (pxDefaultAttribute, hasPersistingCheckNothing);
+        }
+
+        private static AttributeData GetPXDefaultAttribute(PXContext pxContext, AttributeInformation attributeInformation,
+            ImmutableArray<AttributeData> attributes, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            return attributes
```

Comment:
@metallmonster, shouldn't we check only the inheritance chain of the `a` for `PXUnboundDefault`? How will the system behave if the  `a` is an aggregator which contains both `PXDefault` and `PXUnboundDefault`?

Reply:
@SENya1990 the method searches for `PXDefaultAttribute` and its derived classes. We need to exclude `PXUnboundDefaultAttribute` and its derived classes since it is not covered by this analyzer.
The system is not intended to behave if we have both attributes because it is inconsistent situation.